internal/venice-common/src/main/java/com/linkedin/venice/kafka/consumer/ApacheKafkaConsumer.java

```diff
@@ -136,6 +128,7 @@ public void batchUnsubscribe(Set<PubSubTopicPartition> pubSubTopicPartitionSet)
     Set<TopicPartition> newTopicPartitionAssignment = new HashSet<>(kafkaConsumer.assignment());
     newTopicPartitionAssignment.removeAll(convertPubSubTopicPartitionSetToTopicPartitionSet(pubSubTopicPartitionSet));
     kafkaConsumer.assign(newTopicPartitionAssignment);
+    assignments.removeAll(pubSubTopicPartitionSet);
```

Comment:
Yeah, that would be nicer. I planned to change `ApacheKafkaConsumer#assignments` into mapping from `TopicPartition` to `PubSubTopicPartition` Map to avoid using `PubSubTopicRepository`. But your idea still works here.